src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.service;
+
+import org.apache.commons.collections4.CollectionUtils;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.ccd.items.DepositTypeItem;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.CLAIMANT_TITLE;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.DEPOSIT_REFUNDED_GREATER_DEPOSIT_ERROR;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.RESPONDENT_TITLE;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.UNABLE_TO_FIND_PARTY;
+
+public class DepositOrderValidationService {
+
+    private DepositOrderValidationService() {
+    }
+
+    public static List<String> validateDepositOrder(CaseData caseData) {
+        List<String> errors = new ArrayList<>();
+        if (!CollectionUtils.isEmpty(caseData.getDepositCollection())) {
+
+            for (DepositTypeItem depositTypeItem : caseData.getDepositCollection()) {
+                if (!isNullOrEmpty(depositTypeItem.getValue().getDepositAmountRefunded())
+                        && (isNullOrEmpty(depositTypeItem.getValue().getDepositAmount())
+                        || Integer.parseInt(depositTypeItem.getValue().getDepositAmountRefunded())
+                        > Integer.parseInt(depositTypeItem.getValue().getDepositAmount()))) {
+                    errors.add(DEPOSIT_REFUNDED_GREATER_DEPOSIT_ERROR);
+                }
+
+                validateDepositOrderAgainst(errors, depositTypeItem);
+                validateDepositRequestedBy(errors, depositTypeItem);
+                validateDepositRefunded(errors, depositTypeItem);
+            }
+        }
+        return errors;
+    }
+
+    public static void validateDepositRefunded(List<String> errors, DepositTypeItem depositTypeItem) {
```

Comment:
Why is this method and the following methods public?

Reply:
Refactored